### PR TITLE
Fetch latest internship repo

### LINKED ISSUE #34: Dynamically Fetch Latest Internship Repository
**Is your feature request related to a problem? Please describe.**
Currently, the repository for internship postings (e.g., `SimplifyJobs/Summer2025-Internships`) is hardcoded. Since the repository name changes every year (e.g., `Summer2025-Internships` → `Summer2026-Internships`), this hardcoding could cause the bot to fail when the repository is updated.

To ensure reliability, we need to implement a mechanism that dynamically fetches the latest repository by querying the GitHub API.

**Describe the solution you'd like**
1. Use the GitHub API to search for repositories in the SimplifyJobs organization that follow the pattern `SummerYYYY-Internships`.
1. Determine the latest repository based on the year in its name.
1. Update the bot initialization to use the dynamically fetched repository name.

**Describe alternatives you've considered**
N/A

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by aarontorres0**
Resolves #34

### CODE REVIEW

File: src/DiscordBot.py [hunk 1]
```diff
@@ -206,13 +207,42 @@ async def on_ready():
     """
     logger.info(f"Logged in as {bot.user.name}")
 
-    github_internship = GitHubUtilities(
-        token=GITHUB_TOKEN, repo_name="SimplifyJobs/Summer2025-Internships", isSummer=True, isCoop=True
+    try:
+        latest_repo = get_latest_internship_repo()
```
Comment (DavidSalazar123) [inline src/DiscordBot.py:211]: This should not be called here. This is only triggered when the bot is setting up. It should be in `scheduled_task` when creating a github connection

File: src/DiscordBot.py [hunk 2]
```diff
@@ -206,13 +207,42 @@ async def on_ready():
...
+            token=GITHUB_TOKEN, repo_name=f"SimplifyJobs/{latest_repo}", isSummer=True, isCoop=True
+        )
+        github_newgrad = GitHubUtilities(token=GITHUB_TOKEN, repo_name="SimplifyJobs/New-Grad-Positions")
+        job_utilities = JobsUtilities()
+        scheduled_task.start(github_internship, github_newgrad, job_utilities)  # Start the loop
+        health_check_task.start()  # Start health check task
+    except Exception as e:
+        logger.error(f"Failed to initialize bot with latest repository: {e}", exc_info=True)
+
+def get_latest_internship_repo():
```
Comment (DavidSalazar123) [inline src/DiscordBot.py:224]: This method should not exist exist within main.py. It should exist within JobsUtilities.py

File: src/DiscordBot.py [hunk 3]
```diff
@@ -206,13 +207,42 @@ async def on_ready():
...
+    g = Github(GITHUB_TOKEN)
+    org = g.get_organization("SimplifyJobs")
+    repos = org.get_repos()
+
+    matching_repos = []
+    for repo in repos:
+        if repo.name.startswith("Summer"):
```
Comment (DavidSalazar123) [inline src/DiscordBot.py:231]: What about NewGrad?

File: src/DiscordBot.py [hunk 4]
```diff
@@ -206,13 +207,42 @@ async def on_ready():
...
+            if len(year) == 4 and year.isdigit():
+                matching_repos.append(repo.name)
+
+    if not matching_repos:
+        raise ValueError(
+            "No repositories matching the pattern 'SummerYYYY-Internships' were found in the organization SimplifyJobs. Make sure the naming format hasn't changed and that your GitHub token has the necessary permissions."
+        )
+
+    sorted_repos = sorted(
+        matching_repos, key=lambda name: int(name[len("Summer"):].split("-")[0]), reverse=True
```
Comment (DavidSalazar123) [inline src/DiscordBot.py:243]: Why do you sort when you can just do max here?

**Review by DavidSalazar123 — COMMENTED** [review]
Please ensure the test cases pass as they are now set up correctly

File: src/DiscordBot.py (hunk 3 above)
Comment (aarontorres0) [reply src/DiscordBot.py:231]: The new grad repo doesn't have a year as part of the naming convention. The repo `SimplifyJobs/New-Grad-Positions` has the new grad positions for 2025.

File: src/DiscordBot.py (hunk 4 above)
Comment (aarontorres0) [reply src/DiscordBot.py:243]: Good catch

File: src/DiscordBot.py (hunk 2 above)
Comment (aarontorres0) [reply src/DiscordBot.py:224]: I agree 👍

**Comment by aarontorres0** [comment]
> Please ensure the test cases pass as they are now set up correctly

Some of the tests were outdated so I made some changes to them

File: src/config.py
Comment (DavidSalazar123) [inline src/config.py:1]: Why is this here? Please remove this config.py file as we already have .env file that is leveraged

File: src/JobsUtilities.py [hunk 5]
```diff
@@ -195,3 +197,26 @@ async def getJobs(
...
+            if len(year) == 4 and year.isdigit():
+                matching_repos.append(repo.name)
+
+    if not matching_repos:
+        raise ValueError(
+            "No repositories matching the pattern 'SummerYYYY-Internships' were found in the organization SimplifyJobs. Make sure the naming format hasn't changed and that your GitHub token has the necessary permissions."
+        )
+
+    latest_repo = max(
+        matching_repos, key=lambda name: int(name[len("Summer"):].split("-")[0])
```
Comment (DavidSalazar123) [inline src/JobsUtilities.py:220]: Do we still need to sort here? Wouldn't the max return from [2024Internships, 2025Internships] just be 2025Internships?

**Comment by DavidSalazar123** [comment]
I understand why you might want the `config.py` but it doesn't help in our situation as there isn't any [conditionals](https://softwareengineering.stackexchange.com/questions/351126/how-bad-of-an-idea-is-it-to-use-python-files-as-configuration-files) that affect the config set up

File: src/JobsUtilities.py (hunk 5 above)
Comment (aarontorres0) [reply src/JobsUtilities.py]: No, this would not include `Internships`. It would just be the year `20XX` part

File: src/DiscordBot.py [hunk 6]
```diff
@@ -71,21 +71,25 @@ async def health_check_task():
...
     Parameters:
-        - internship_github: Internship object
-        - newgrad_github: New Grad GitHubUtilities object
         - job_utilities: The JobsUtilities object
     """
     async with lock:
         try:
             start_time = datetime.now()
 
+            latest_repo = JobsUtilities.get_latest_internship_repo()
```
Comment (DavidSalazar123) [inline src/DiscordBot.py:85]: Still concerned about this line here. Why do we need to make an api call every minute to check the latest repo? Why not first try to make a connection to latest cached repo and if it doesn't exist, loop until we can find the latest repo. Thoughts?

Comment (aarontorres0) [reply src/DiscordBot.py:85]: I think that's reasonable. I'll see what I can do

Comment (DavidSalazar123) [reply src/DiscordBot.py:85]: Could you explain further?

Comment (aarontorres0) [reply src/DiscordBot.py:85]: I agree that it is reasonable to try to save the repo somewhere and only get the latest repo whenever the one we have saved is outdated. In my most recent commit (c8483df2ad57bf786eb930da467aca153ccacb88) I save the repo and always try to use what I have saved in the variable. If the repo we have saved no longer works than I fetch the latest repo again

**Comment by DavidSalazar123** [comment]
@aarontorres0 Move your test_repo.py into test folder and I will merge!

**Comment by aarontorres0** [comment]
> @aarontorres0 Move your test_repo.py into test folder and I will merge!

That was a file I was using for testing some methods locally. Let me remove it

**Review by DavidSalazar123 — APPROVED** [review]
LGTM
